core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java

```diff
@@ -72,7 +72,7 @@ public String getName()
     {
         Session session = stateMachine.getSession();
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getName());
-        Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+        Optional<TableHandle> tableHandle = metadata.getOriginalTableHandle(session, tableName, Optional.of(getName()));
```

Comment:
> Fail modifications for redirected tables in engine

I am not convinced we should to that

Reply:
here "modifications" is bad wording, meant DDLs. we already added support for DMLs in https://github.com/trinodb/trino/pull/8683

The last discussion that we had on this was https://github.com/trinodb/trino/pull/7606#issuecomment-843542294

> DDL operations are problematic since users need to be aware of which connector they are using, since they have different data types, partitioning, bucketing, etc. Our thought was that these are relatively rare operations by more advanced users and that trying to have hidden redirections would end up causing more confusion.

Reply:
do you also feel otherwise for DDLs?

Reply:
do you also feel otherwise for DDLs?

Reply:
Users sooner or later will ask why `ALTER ... ADD COLUMN (something varchar)` does not work, and I won't be able to explain to them why not. Yes, from engineer perspective, we may have harder time supporting things like column properties (maybe, or maybe not), but i would assume there are not always used, so users who do not intend to use column properties, will not accept this as a rational explanation for the limitation.
TL;DR yes, DDLs like ALTER .. ADD/DROP COLUMN should be routed as well.

cc @losipiuk @alexjo2144